src/Compilers/VisualBasic/Portable/Utilities/FeatureUtils.vb

```diff
@@ -0,0 +1,84 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
+    ''' <summary>
+    ''' Utility methods to help with checking the availability of language features.
+    ''' </summary>
+    Friend Module FeatureUtils
+
+        ''' <summary>
+        ''' Check to see if the given <paramref name="feature"/> is available with the <see cref="LanguageVersion"/>
+        ''' of the parser.
+        ''' </summary>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="options">The parse options being used.</param>
+        ''' <returns>True if the feature's language version is compatible with the specified language version.</returns>
+        <Runtime.CompilerServices.Extension>
+        Private Function IsInLanguageVersion(feature As Feature, options As VisualBasicParseOptions) As Boolean
+            Dim required = feature.GetLanguageVersion()
+            Dim current = options.LanguageVersion
+            Return required <= current
+        End Function
+
+        <Runtime.CompilerServices.Extension>
+        Private Function CheckFeatures(feature As Feature, options As VisualBasicParseOptions) As Boolean
```

Comment:
@cston Keeping this as it checking the `VisualBasicParserOptions.Features`, easier to see that there two distinct checks;- Features and Language Version.